etcdctl/ctlv3/command/printer_simple.go

```diff
@@ -141,7 +141,11 @@ func (s *simplePrinter) RoleGet(role string, r v3.AuthRoleGetResponse) {
 	printRange := func(perm *v3.Permission) {
 		sKey := string(perm.Key)
 		sRangeEnd := string(perm.RangeEnd)
-		fmt.Printf("\t[%s, %s)", sKey, sRangeEnd)
+		if strings.Compare(sRangeEnd, "\xff") != 0 {
```

Comment:
Yes, my understanding of Range interface was very wrong... any byte sequences (other than ascii strings) are allowed to be a key so 0xff cannot be the largest key. I'll follow your suggestions and use `StringAffineComparable` for the purpose.